velox/common/base/Fs.h

```diff
@@ -15,10 +15,32 @@
  */
 #pragma once
 
+#include <optional>
+
 #if __has_include("filesystem")
 #include <filesystem>
 namespace fs = std::filesystem;
 #else
 #include <experimental/filesystem>
 namespace fs = std::experimental::filesystem;
 #endif
+
+namespace facebook::velox::common {
+
+/// Generates a file directory specified by 'dirPath'. The generation will be
+/// recursive. Non-exist parent directories will also be created. Returns true
+/// if creation is successful, false otherwise. Error message will be printed if
+/// creation is unsuccessful, but already created directories will not be
+/// removed.
+bool generateFileDirectory(const char* dirPath);
+
+/// Creates a file with a generated file name in provided 'basePath'. The
+/// generated file will have random chars in the file name to avoid duplication.
+/// The full path of the file will be of the pattern
+/// {basePath}/velox_{prefix}_XXXXXX where 'XXXXXX' is the randomly generated
+/// chars. A nullopt will be returned if file creation fails.
+std::optional<std::string> generateFilePath(
```

Comment:
Did you look for similar functionality under std::filesystem as well? Looks like something that should be provided by that library.

Reply:
I have not. This method is moved from VectorSavor to Fs.h as its correct place to stay. I can take a look though.

Reply:
I have checked and the only promising one is std::filesystem::temp_directory_path() which only works for directory and does not provide base path functionality.